applications/stepswitch/src/stepswitch_util.erl

```diff
@@ -49,11 +50,48 @@ get_realm(JObj) ->
 -spec get_inbound_destination(kz_json:object()) -> kz_term:ne_binary().
 get_inbound_destination(JObj) ->
     {Number, _} = kapps_util:get_destination(JObj, ?APP_NAME, <<"inbound_user_field">>),
-    case kapps_config:get_is_true(?SS_CONFIG_CAT, <<"assume_inbound_e164">>, 'false') of
+    convert_to_e164_format(Number, ?SS_CONFIG_CAT, <<"assume_inbound_e164">>).
+
+%%------------------------------------------------------------------------------
+%% @doc Convert number to e164 format depending on config values set in system_config db
+%% @end
+%%------------------------------------------------------------------------------
+-spec convert_to_e164_format(kz_term:ne_binary(), kz_term:ne_binary(), kz_term:ne_binary()) -> kz_term:ne_binary().
+convert_to_e164_format(Number, ConfigDoc, ConfigKey) ->
+    case kapps_config:get_is_true(ConfigDoc, ConfigKey, 'false') of
         'true' -> assume_e164(Number);
         'false' -> knm_converters:normalize(Number)
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc Get the origination DID of a inbound call if its a number, else return undefined
+%% @end
+%%------------------------------------------------------------------------------
+-spec get_inbound_origination(kz_json:object()) -> kz_term:ne_binary().
+get_inbound_origination(JObj) ->
+    {Number, _} = kapps_util:get_origination(JObj, ?APP_NAME, <<"inbound_user_origination_field">>),
+    case verify_is_did(Number) of
+        'true' ->
+            convert_to_e164_format(Number, ?SS_CONFIG_CAT, <<"assume_inbound_origination_e164">>);
+        'false' ->
+            'undefined'
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc Verify the DID is a number or a number beginning with +
+%% @end
+%%------------------------------------------------------------------------------
+-spec verify_is_did(kz_term:ne_binary()) -> boolean().
+verify_is_did(<<$+, Did/binary>>) ->
```

Comment:
@bradfordben Above I am recommending removing this but I thought it worth mentioning that if the number starts with a plus then it was provided in e164 format.  In that case it would be preferable to return as is and not attempt a double conversion... IE:

verify_is_did(<<$+, _/binary>> = E164) -> E164;

Reply:
Good point 👍